### PR TITLE
school library: unit tests

### PR DESCRIPTION
**Pull request by Zeph180**
1. Add unit tests for all the methods in all the classes
- Book
- Person
- Student
- Teacher
- Classroom
- Rental

### CODE REVIEW

File: spec/book_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require 'rspec'
+require_relative '../book'
+
```
Comment (KDlamini) [inline spec/book_spec.rb]: As per project requirements, kindly make sure that there are unit tests for all the methods in all the classes.

- Please test the `add_rental` method in the `Book` class.

Here's an example of how you can achieve this:

```ruby
context "When adding a new rental" do
    it "The add_rental method returns a rental and adds it to the book" do
      book = Book.new("The Hunger Games", "Suzanne Collins")
      person = Person.new(24, "Tomas")

      rental = book.add_rental("2021-05-04", person)

      expect(book.rentals).to eql [rental]
    end
end
```

File: spec/classroom_spec.rb [hunk 2]
```diff
@@ -0,0 +1,17 @@
+require 'rspec'
+require_relative '../classroom'
+require_relative '../student'
+
```
Comment (KDlamini) [inline spec/classroom_spec.rb]: Same here, please test the `add_student` method in the `Classroom` class.

- Consider the example I mentioned above on how you can achieve this.

File: spec/person_spec.rb [hunk 3]
```diff
@@ -0,0 +1,56 @@
+require 'rspec'
+require_relative '../person'
+
```
Comment (KDlamini) [inline spec/person_spec.rb]: When I run the tests I get an error.

image

- Kindly remove the `require 'rspec'` in all the test files. There's no need to require rspec in order to run tests in your project. You simply need to add `gem 'rspec-rails', '~> 6.0.0'` to the `Gemfile`, Then run `bundle install` to install it in your project.
- Ruby will automatically require rspec in all test files ending with `*_spec.rb`.

## Summary

After I removed the `require 'rspec'`, the tests passed. Great job !!! 💯 

image

**Review by KDlamini — CHANGES_REQUESTED** [review]
### Hi team @Zeph180, @djo1975 🙋‍♀️,  

Great work so far! 💯 
You have done a great job writing the unit tests for the OOP School Library. 🎊 
There are some issues that you still need to work on to improve your project according to the requirements. You are almost there! 💪🏾 

## Highlights 🌞
 
✔️  Good PR title and description
✔️  All test files are placed in a different directory (e.g `/spec`)
✔️  There is one test file per class
✔️  There is at least an 'it' block for each class method

## Required Changes ♻️
 
**Kindly look through the comments below 👇 for required changes**

Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/book_spec.rb (hunk 1 above)
Comment (KDlamini) [reply spec/book_spec.rb]: Well resolved ✔️

File: spec/classroom_spec.rb (hunk 2 above)
Comment (KDlamini) [reply spec/classroom_spec.rb]: Well resolved ✔️

File: spec/person_spec.rb (hunk 3 above)
Comment (KDlamini) [reply spec/person_spec.rb]: Well resolved ✔️

**Review by KDlamini — APPROVED** [review]
### Hi team @Zeph180, @djo1975 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ All public methods in each class are well tested

# Status: Approved ✔️ 

oscars-standing-ovation
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by djo1975** [comment]
Thx :)

**Pull request merged**